glossary.md

```diff
@@ -0,0 +1,241 @@
+
+L1 re-orgs can happen because of network conditions or attacks. L2 re-orgs are a
+consequence of L1 re-orgs, mediated via [L2 chain derivation][derivation].
+
+## Precompiled Contract ("Precompile")
+[precompile]: /glossary.md#precompiled-contract-precompile
+
+A contract that exists on chain but consumes no storage, and whose invocation
+does not need to result in the execution of EVM bytecode. Instead, its behaviour
+is defined by specification.
+
+Ethereum L1 precompiles are specified in Appendix E of the [yellow
+paper][yellow].
+
+Optimistic Ethereum exposes all of Ethereum's precompiles, and adds the
+following precompiles:
+
+- [L1 Attributes Precompiled Contract][l1-attr-precompile]
+
+--------------------------------------------------------------------------------
+
+# L2 Chain Concepts
+
+## L2 Chain Inception
+[L2 chain inception]: /glossary.md#L2-chain-inception
+
+The L1 block number for which the first block of the L2 chain was generated.
+
+## Rollup Node
+[rollup node]: /glossary.md#rollup-node
+
+The rollup node is responsible for [deriving the L2 chain][derivation] from the
+[L2 derivation inputs][deriv-inputs] available on L1. This is done by its
+[rollup driver] component.
+
+The rollup node is also responsible for [block gossip], the process of
+transmitting the L2 blocks it derives over a peer-to-peer network.
+
+- cf. [Rollup Node Specification](/rollup-node.md)
+- cf. [Block Gossip Specification](TODO)
+
+> **TODO LINK** block gossip spec
+
+## Rollup Driver
+[rollup driver]: /glossary.md#rollup-driver
+
+The rollup driver is the [rollup node] component responsible for [deriving the
+L2 chain][derivation] from the [L2 derivation inputs][deriv-inputs] available on
+L1.
+
+## L2 Chain Derivation
+[derivation]:  /glossary.md#L2-chain-derivation
+
+A process that reads [L2 derivation inputs][deriv-inputs] from L1 in order to
+derive the L2 chain.
+
+cf. [L2 Chain Derivation (in Rollup Node
+Specification)](/rollup-node.md#l2-chain-derivation)
+
+## L2 Derivation Inputs
+[deriv-inputs]: /glossary.md#l2-chain-derivation-inputs
+
+This term refers to data that is found in L1 blocks and is read by the [rollup
+node] to construct [payload attributes].
+
+Chain derivation attributes include:
+- L1 block attributes
+   - block number
+   - timestamp
+   - basefee
+- [deposits]
+
+## Payload Attributes
+[payload attributes]: /glossary.md#payload-attributes
+
+This term refers to data that can be derived from [L2 chain derivation
+inputs][deriv-inputs] found on L1, which are then passed to the [execution
+engine] to construct L2 blocks.
+
+"Payload attributes" is a term that originates and is specified in the [Ethereum
+Engine API specification][engine-api].
+
+In addition to the [specified][engine-api] `timestamp`, `random`and
+`suggestedFeeRecipient` fields, we add a field to specify the transactions that
+the execution engine must include in the block and execute.
+
+> **TODO** The transaction field must be specified.
```

Comment:
This is just the glossary, not the full specification. I would reference the engine API spec instead for the details.